src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -299,6 +296,9 @@ def _filter_modname(extensions):
 
                 for ext in allowed_extensions:
                     try:
+                        # Import in the `for` loop because `allowed_extensions` can be []. In such case we
+                        # don't need to import `check_version_compatibility` at all.
+                        from azure.cli.core.extension.operations import check_version_compatibility
```

Comment:
Importing a module in a for loop is not a good practice.
How much can it benefit from importing in this for loop?

Reply:
You can import module as many times as you want in one Python program, no matter what module it is. Every subsequent import after the first accesses the cached module instead of re-evaluating it. So there is no performance penalty.

---
In reply to: [454774982](https://github.com/Azure/azure-cli/pull/14331#discussion_r454774982) [](ancestors = 454774982)

Reply:
This is a tricky part. I just want to eliminate an extra `if allowed_extensions:`. Python has an [import cache](https://docs.python.org/3/reference/import.html#the-module-cache) which will handle "import in a for loop" nicely.